tracer/build/_build/Build.Gitlab.cs

```diff
@@ -102,9 +104,44 @@ void SignBinary(AbsolutePath binaryPath)
                     logOutput: false,
                     logInvocation: false);
             signProcess.WaitForExit();
+
+            var output = signProcess.Output.Select(o => o.Text);
+            foreach (var line in output)
+            {
+                Logger.Information("[dd-wcs] {Line}", line);
+
+                // dd-wcs will return 0 even if there are errors
+                if (line.StartsWith("ERROR:", StringComparison.OrdinalIgnoreCase))
+                {
+                    throw new Exception($"Error found when signing {binaryPath}: {line}");
+                }
+            }
+
             if (signProcess.ExitCode == 0)
             {
-                PowerShellTasks.PowerShell($"Get-AuthenticodeSignature {binaryPath}");
+                var status = PowerShellTasks.PowerShell(
+                    $"(Get-AuthenticodeSignature '{binaryPath}').Status",
+                    logOutput: false,
+                    logInvocation: false);
+
+                var lines = status.Select(o => o.Text?.Trim()).Where(l => !string.IsNullOrEmpty(l)).ToList();
+
+                if (lines.Count == 0 || !lines[0].Equals("Valid", StringComparison.OrdinalIgnoreCase))
+                {
+                    throw new Exception($"Signature verification failed for {binaryPath}. Status: {(lines.Count > 0 ? lines[0] : "Empty")}");
+                }
```

Comment:
You can simplify this a little and remove the checks for `lines.Count` by using `FirstOrDefault()`. Something like:

```csharp
var result = status.Select(o => o.Text)
                   .FirstOrDefault(l => !string.IsNullOrEmpty(l))
                  ?.Trim();

if (!string.Equals(result, "Valid", StringComparison.OrdinalIgnoreCase))
{
    throw new Exception($"Signature verification failed for {binaryPath}. Status: {result ?? "Empty"}");
}
```

Same comment below for the second command.